File: constrained-device-app/src/main/python/programmingtheiot/cda/system/SystemMemUtilTask.py
#####
# 
# This class is part of the Programming the Internet of Things project.
# 
# It is provided as a simple shell to guide the student and assist with
# implementation for the Programming the Internet of Things exercises,
# and designed to be modified by the student as needed.
#

import logging
import psutil

from programmingtheiot.cda.system.BaseSystemUtilTask import BaseSystemUtilTask
import programmingtheiot.common.ConfigConst as ConfigConst

class SystemMemUtilTask(BaseSystemUtilTask):
	"""
	Shell representation of class for System Memory Utility.
	
	"""

	def __init__(self):
		"""
		Constructor
		"""
		super(SystemMemUtilTask, self).__init__(sensorName = ConfigConst.MEM_UTIL_NAME)
	
	def _getSystemUtil(self) -> float:
		"""
		Get the used system memory percentage
		
		@return float
		"""
		return psutil.virtual_memory().percent
		

File: gateway-device-app/src/main/java/programmingtheiot/gda/system/SystemMemUtilTask.java
/**
 * This class is part of the Programming the Internet of Things project.
 * 
 * It is provided as a simple shell to guide the student and assist with
 * implementation for the Programming the Internet of Things exercises,
 * and designed to be modified by the student as needed.
 */ 

package programmingtheiot.gda.system;

import java.lang.management.ManagementFactory;
import java.lang.management.MemoryUsage;
import java.util.logging.Logger;

/**
 * Shell representation of class for student implementation.
 * 
 */
public class SystemMemUtilTask extends BaseSystemUtilTask
{
	// constructors
	
	/**
	 * Default.
	 * 
	 */
	public SystemMemUtilTask()
	{
		super();
	}
	private static final Logger _Logger =
			Logger.getLogger(BaseSystemUtilTask.class.getName());
	
	// protected methods
	
	@Override
	protected float getSystemUtil()
	{
		MemoryUsage memUsage = ManagementFactory.getMemoryMXBean().getHeapMemoryUsage();
		 _Logger.info("Memory used: " + memUsage.getUsed());
	
		 _Logger.info("Memory avail: " + memUsage.getMax());
		 double val = ((double) memUsage.getUsed() / (double) memUsage.getMax()) * 100.0d;
		 _Logger.info("Memory used %: " + val);
		return (float) val;
	}
	
}
